src/python/pants/bin/local_pants_runner.py

```diff
@@ -220,11 +221,22 @@ def set_start_time(self, start_time):
     self._run_start_time = start_time
     self._reporting.initialize(self._run_tracker, self._options, start_time=self._run_start_time)
 
+    target_specs = [self.normalise_spec(spec) for spec in self._options.target_specs]
+    self._run_tracker.run_info.add_info("specs", target_specs, stringify=False)
```

Comment:
Why would we skip stringifying here? It's not clear that we need to, unless we haven't correctly made these into strings ourselves before calling this.

Reply:
Something I learnt through this change is that we currently convert all values here to strings by calling `str(_)`, rather than something JSON-aware, which means that for a `list`, we would get a value:

`{"key": "['val', 'val']"}`

where what we actually want is:

`{"key": ["val", "val"]}`

I think the presence of the `str` call is a bug (or rather, over-broad - it should only apply to types which can't be JSON stringified on their own). But backwards compatibility is backwards compatibility...